src/main/java/com/ibm/crypto/plus/provider/CurveUtil.java

```diff
@@ -21,38 +20,31 @@ public enum CURVE {
         X25519, X448, FFDHE2048, FFDHE3072, FFDHE4096, FFDHE6144, FFDHE8192, Ed25519, Ed448
     }
 
-    private static HashMap<CURVE, Integer> publicCurveSizes = new HashMap<CURVE, Integer>(); // key sizes of curves in bytes
-    private static HashMap<Integer, CURVE> sizesToPublicCurves = new HashMap<Integer, CURVE>(); // reverse of above
-    private static HashMap<CURVE, Integer> privateCurveSizes = new HashMap<CURVE, Integer>(); // key sizes of curves in bytes
-    private static HashMap<Integer, CURVE> sizesToPrivateCurves = new HashMap<Integer, CURVE>(); // reverse of above
+    private static HashMap<CURVE, Integer> curveSizes = new HashMap<CURVE, Integer>(); // key sizes of curves in bytes
+    private static HashMap<CURVE, Integer> DEREncodingSizes = new HashMap<CURVE, Integer>(); // key sizes of der encoded private key values.
     private static HashMap<Integer, CURVE> sizesToCurves = new HashMap<Integer, CURVE>(); // maps the total key size (I think?)
                                                                                           // to algorithm (used in constructor)
     static {
-        publicCurveSizes.put(CURVE.X25519, 32);
-        publicCurveSizes.put(CURVE.X448, 56);
-        publicCurveSizes.put(CURVE.FFDHE2048, 557);
-        publicCurveSizes.put(CURVE.FFDHE3072, 813);
-        publicCurveSizes.put(CURVE.FFDHE4096, 1069);
-        publicCurveSizes.put(CURVE.FFDHE6144, 1581);
-        publicCurveSizes.put(CURVE.FFDHE8192, 2093);
-        publicCurveSizes.put(CURVE.Ed25519, 32);
-        publicCurveSizes.put(CURVE.Ed448, 57);
 
-        for (Map.Entry<CURVE, Integer> entry : publicCurveSizes.entrySet())
-            sizesToPublicCurves.put(entry.getValue(), entry.getKey());
-
-        privateCurveSizes.put(CURVE.X25519, 48);
-        privateCurveSizes.put(CURVE.X448, 72);
-        privateCurveSizes.put(CURVE.FFDHE2048, 327);
-        privateCurveSizes.put(CURVE.FFDHE3072, 461);
-        privateCurveSizes.put(CURVE.FFDHE4096, 595);
-        privateCurveSizes.put(CURVE.FFDHE6144, 857);
-        privateCurveSizes.put(CURVE.FFDHE8192, 1117);
-        privateCurveSizes.put(CURVE.Ed25519, 32);
-        privateCurveSizes.put(CURVE.Ed448, 57);
-
-        for (Map.Entry<CURVE, Integer> entry : privateCurveSizes.entrySet())
-            sizesToPrivateCurves.put(entry.getValue(), entry.getKey());
+        curveSizes.put(CURVE.X25519, 32);
+        curveSizes.put(CURVE.X448, 56);
+        curveSizes.put(CURVE.FFDHE2048, 557);
```

Comment:
I'm not sure we're even using those, but the numbers seem a bit off. Should we check what the expectation is?

Reply:
I did research some the sizes of keys for X25519, X448, and FFDHE2048. Teh x22519 and X448 values seemed reasonable to me and were in use prior to this update. The FFDHE2048 size I also didnt change for this update. Do you see a path here that may be causing issues with the 557 size? That 557 size does seem odd to me as i would expect 2048 for the key size instead.

Reply:
I updated the keysizes to match both of our expectations here for the key size pattern.

Reply:
Hi @KostasTsiounis I did comment out the `DEREncodingSizes` for FFDHE as they are not in use. I did however find that I need to keep the values for `curveSizes` associated with FFDHE curves. Please let me know if this looks ok to you.

Reply:
LGTM